File: moneyverse-be-bom/moneyverse-be-lib/moneyverse-user-management/src/main/java/it/moneyverse/user/services/UserService.java
package it.moneyverse.user.services;

import it.moneyverse.core.model.dto.UserDto;
import it.moneyverse.user.model.dto.LanguageDto;
import it.moneyverse.user.model.dto.UserUpdateRequestDto;
import java.util.List;
import java.util.UUID;

public interface UserService {

  UserDto getUser(UUID userId);

  UserDto updateUser(UUID userId, UserUpdateRequestDto request);

  void disableUser(UUID userId);

  void deleteUser(UUID userId);

  void checkIfUserExists(UUID userId);

  List<LanguageDto> getLanguages();
}

File: moneyverse-fe/src/app/shared/services/user.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../../environments/environment';
import {UserUpdateRequestDto} from '../../core/auth/models/user.model';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private readonly httpClient = inject(HttpClient);

  public updateUser(userId: string, request: Partial<UserUpdateRequestDto>) {
    return this.httpClient.put(`/users/${userId}`, request);
  }

  public deleteUser(userId: string) {
    return this.httpClient.delete(`/users/${userId}`);
  }
}
